app/code/Magento/Backend/Controller/Adminhtml/Index/GlobalSearch.php

```diff
@@ -6,6 +6,11 @@
  */
 namespace Magento\Backend\Controller\Adminhtml\Index;
 
+use Magento\Backend\Model\Search\ItemsFactory;
+
+/**
+ * @api
```

Comment:
Current Controller became big enough and now contains Business logic inside.
It's better to introduce dedicated handler/model for this.

In M2 we are trying to make Controllers as light/ thin as possible

This Handler would be marked as @ api intead of current controller